action.yml

```diff
@@ -1,15 +1,148 @@
-name: 'Rainforest QA: RFML Runner'
-description: 'Run RFML tests from your repo'
+name: Rainforest QA GitHub Action
+description: Easily trigger a Rainforest Run from your GitHub workflows
 inputs:
-  rainforest_api_key:  # id of input
-    description: 'Rainforest API key'
+  description:
+    description: An arbitrary string to associate with the run
+    required: false
+    default: ''
+
+  run_group_id:
+    description: Only run tests tied to this Run Group
     required: true
+
+  environment_id:
+    description: Use a specific environment for this run
+    required: false
+    default: ''
+
+  conflict:
+    description: How other currently in-progress runs should be handled. Values are `abort` to cancel runs in the same environment as your new run and `abort-all` to cancel all runs
+    required: false
+    default: ''
+
+  crowd:
+    description: The crowd to use for this run
+    required: false
+    default: ''
+
+  release:
+    description: Manually entered release information about the release the run is associated with
+    required: false
+    default: ''
+
+  token:
+    description: Your Rainforest QA API token
+    required: true
+
+  dry_run:
+    description: Set to true to run parameter validations without starting a new Rainforest run
+    required: false
+    default: ''
+
+  background:
+    description: Do not wait for a run to complete before exiting
+    required: false
+    default: ''
 runs:
-  using: 'docker'
-  image: 'docker://gcr.io/rf-public-images/rainforest-cli:latest'
-  args:
-    - 'run'
-    - '--fail-fast'
-    - '-f'
-    - 'spec/rainforest'
+  using: composite
+  steps:
+    - name: Set Action Version
+      shell: bash
+      run: |
+        echo "version=0.0.1" >> $GITHUB_ENV
+    - name: Validate Parameters
+      shell: bash
+      id: validate
+      run: |
+        # Show Action Version
+        echo "Using Rainforest GitHub Action v${{ env.version }}"
+
+        # Ensure results directory is there
+        mkdir -p results/rainforest
+
+        echo "::add-mask::${{ inputs.token }}"
+
+        # Validate token
+        if [ -z "${{ inputs.token }}" ] ; then
+          echo "Error: Token not set"
+          exit 1
+        fi
+
+        # Validate run_group_id
+        if ! echo "${{ inputs.run_group_id }}" | grep -Eq '^[0-9]+$' ; then
+          echo "Error: run_group_id not a positive integer (${{ inputs.run_group_id }})"
+          exit 1
+        fi
+
+        RUN_COMMAND="run --skip-update --token ${{ inputs.token }} --run-group ${{ inputs.run_group_id }} --junit-file results/rainforest/junit.xml"
+
+        # Validate conflict
+        if [ -n "${{ inputs.conflict }}" ] ; then
+          if echo "abort abort-all" | grep -Eq "\b"${{ inputs.conflict }}"\b" ; then
```

Comment:
Maybe prefer a `case` statement so we don't need the regex?

Untested:
```
case "${{ inputs.conflict }}")
  abort)
  abort-all)
    RUN_COMMAND="${RUN_COMMAND} --conflict ${{ inputs.conflict }}"
  ;;
  *)
    echo "Error: ${{ inputs.conflict }} not in (abort abort-all)"
    exit 1
  ;;
esac
```

Reply:
👍🏼 with a couple fixes (`case ... in` and `abort) ;&` to fall through)